addons/web/static/src/core/utils/autoresize.js

```diff
@@ -0,0 +1,74 @@
+/** @odoo-module **/
+
+import { useRefListener } from "./hooks";
+import { useEffect } from "@odoo/owl";
+
+/**
+ * This is used on text inputs or textareas to automatically resize it based on its
+ * content each time it is updated. It takes the reference of the element as
+ * parameter and some options. Do note that it may introduce mild performance issues
+ * since it will force a reflow of the layout each time the element is updated.
+ *
+ * @param {Ref} ref
+ */
+export function useAutoresize(ref, options = {}) {
+    useEffect(() => {
```

Comment:
you can simplify the implementation like this:
```js
useEffect(el => { 
  if (el) {
    const resize = ref.el instanceof HTMLInputElement ? resizeInput : resizeTextArea;
    el.addEventListener("input", resize);
    return () => el.removeEventListener("input", resize);
  }
}, [ref.el]);
```
It's slightly simpler, does not need useRefListener, and is more performant

Reply:
Not really sure about how it works but in practice it doesn't